src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -171,6 +172,62 @@ public PackageBuilder GetPackageBuilder(IPackTaskRequest<IMSBuildItem> request)
                     request.RepositoryBranch,
                     request.RepositoryCommit);
             }
+
+            var hasLicenseExpression = !string.IsNullOrEmpty(request.PackageLicenseExpression);
+            var hasLicenseFile = !string.IsNullOrEmpty(request.PackageLicenseFile);
+            if (hasLicenseExpression || hasLicenseFile)
+            {
+                if (string.IsNullOrEmpty(request.LicenseUrl))
+                {
+                    throw new PackagingException(NuGetLogCode.NU5035, string.Format(
+                        CultureInfo.CurrentCulture,
+                        Strings.Licenses_LicenseUrlCannotBeUsedInConjuctionWithLicense));
+                }
+
+                if (hasLicenseExpression && hasLicenseFile)
+                {
+                    throw new PackagingException(NuGetLogCode.NU5033, string.Format(
+                        CultureInfo.CurrentCulture,
+                        Strings.InvalidLicenseCombination,
+                        request.PackageLicenseExpression));
+                }
+
+                var version = GetLicenseExpressionVersion(request);
+
+                if (hasLicenseExpression)
+                {
+                    if (version.CompareTo(LicenseMetadata.CurrentVersion) <= 0)
+                    {
+                        try
+                        {
+                            var expression = NuGetLicenseExpression.Parse(request.PackageLicenseExpression);
+                            builder.LicenseMetadata = new LicenseMetadata(LicenseType.Expression, request.PackageLicenseExpression, expression, version);
+                        }
+                        catch (NuGetLicenseExpressionParsingException e)
+                        {
+                            throw new PackagingException(NuGetLogCode.NU5032, string.Format(
+                                   CultureInfo.CurrentCulture,
+                                   Strings.InvalidLicenseExpression,
+                                   request.PackageLicenseExpression, e.Message) ,
+                                   e);
+                        }
+                    }
+                    else
+                    {
+                        throw new PackagingException(NuGetLogCode.NU5034, string.Format(
+                                   CultureInfo.CurrentCulture,
+                                   Strings.InvalidLicenseExppressionVersion_VersionTooHigh,
+                                   request.PackageLicenseExpressionVersion,
+                                   LicenseMetadata.CurrentVersion));
+                    }
+                }
+
+                if (hasLicenseFile)
+                {
+                    builder.LicenseMetadata = new LicenseMetadata(type: LicenseType.File, license: request.PackageLicenseFile, expression: null, version: version);
```

Comment:
is PackageLicenseFile supposed to be a full path? LicenseMetadata object model should be strict about accepting absolute/relative path.

Here, you probably want to make sure that the targets themselves convert the relative path to an absolute path. Look at https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets#L70 for example.

Reply:
It's not a full path. 
It's the path relative to the nupkg. 

Following the feedback from MSBuild and SDK team we decided to go the https://github.com/NuGet/Engineering/pull/1704/files#diff-459e2f5e463d0ac02e64e36f3d74b407R207 approach. 

```
<PropertyGroup>
    <PackageLicenseFile>LICENSE.txt</PackageLicenseFile>
</PropertyGroup>
 <ItemGroup>
    <None Include="licenses\LICENSE.txt" Pack="true" PackagePath="$(PackageLicenseFile)"/>
</ItemGroup>
```

Reply:
i meant the path in the include string.

Reply:
The None item include should already be handled by the current implementation isn't it?

I'm just piggybacking off of that.